services/iaas/wait/wait.go

```diff
@@ -0,0 +1,152 @@
+package wait
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/oapierror"
+	"github.com/stackitcloud/stackit-sdk-go/core/wait"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaas"
+)
+
+const (
+	CreateSuccess = "CREATED"
+)
+
+// Interfaces needed for tests
+type APIClientInterface interface {
+	GetNetworkAreaExecute(ctx context.Context, organizationId, areaId string) (*iaas.NetworkArea, error)
+	GetProjectRequestExecute(ctx context.Context, projectId string, requestId string) (*iaas.Request, error)
+	GetNetworkExecute(ctx context.Context, projectId, networkId string) (*iaas.Network, error)
+}
+
+// CreateNetworkAreaWaitHandler will wait for network area creation
+func CreateNetworkAreaWaitHandler(ctx context.Context, a APIClientInterface, organizationId, areaId string) *wait.AsyncActionHandler[iaas.NetworkArea] {
+	handler := wait.New(func() (waitFinished bool, response *iaas.NetworkArea, err error) {
+		area, err := a.GetNetworkAreaExecute(ctx, organizationId, areaId)
+		if err != nil {
+			return false, area, err
+		}
+		if area.AreaId == nil || area.State == nil {
+			return false, area, fmt.Errorf("create failed for network area with id %s, the response is not valid: the id or the state are missing", areaId)
+		}
+		if *area.AreaId == areaId && *area.State == CreateSuccess {
+			return true, area, nil
+		}
+		return false, area, nil
```

Comment:
Typically, we return nil in the response if the waiter isn't done yet. Is there something unique going on here?

Reply:
I think this is valid for most `return false,...` statements in this file

Reply:
It was intentional, I think there's no arm in also returning the current state of the object here. What we have documented for this field is `response contains data regarding the current state of the resource targeted by the async action, if applicable`

Reply:
Ok, good to know 😁